game/minecraft_nbt.ksy

```diff
@@ -0,0 +1,221 @@
+        if: not is_root
+        doc: do not read this property, access `type` instead
+      - id: name
+        type: tag_string
+        if: not is_tag_end
+      - id: payload
+        type:
+          switch-on: type
+          cases:
+            'tag::tag_byte': s1
+            'tag::tag_short': s2
+            'tag::tag_int': s4
+            'tag::tag_long': s8
+            'tag::tag_float': f4
+            'tag::tag_double': f8
+            'tag::tag_byte_array': tag_byte_array
+            'tag::tag_string': tag_string
+            'tag::tag_list': tag_list
+            'tag::tag_compound': tag_compound
+            'tag::tag_int_array': tag_int_array
+            'tag::tag_long_array': tag_long_array
+    instances:
+      type:
+        value: 'is_root ? root_type : nested_type'
+      is_tag_end:
+        value: 'type == tag::tag_end'
+  tag_byte_array:
+    -webide-representation: '{len_data:dec} bytes'
+    seq:
+      - id: len_data
+        type: s4
+      - id: data
+        size: len_data
+  tag_string:
+    -webide-representation: '{data}'
+    seq:
+      - id: len_data
+        type: u2
+        doc: unsigned according to https://wiki.vg/NBT#Specification
+      - id: data
+        size: len_data
+        type: str
+  tag_list:
+    -webide-representation: '{num_tags:dec} entries of type {tags_type}'
+    seq:
+      - id: tags_type
+        type: u1
+        enum: tag
+      - id: num_tags
+        type: s4
+      - id: tags
+        type:
+          switch-on: tags_type
+          cases:
+            'tag::tag_byte': s1
+            'tag::tag_short': s2
+            'tag::tag_int': s4
+            'tag::tag_long': s8
+            'tag::tag_float': f4
+            'tag::tag_double': f8
+            'tag::tag_byte_array': tag_byte_array
+            'tag::tag_string': tag_string
+            'tag::tag_list': tag_list
+            'tag::tag_compound': tag_compound
+            'tag::tag_int_array': tag_int_array
+            'tag::tag_long_array': tag_long_array
+        repeat: expr
+        repeat-expr: num_tags
+  tag_compound:
+    -webide-representation: '{dump_num_tags:dec} entries'
+    seq:
+      - id: tags
+        type: named_tag(false)
+        repeat: until
+        repeat-until: _.is_tag_end
+    instances:
+      dump_num_tags:
+        value: 'tags.size - (tags.last.is_tag_end ? 1 : 0)'
+  tag_int_array:
+    -webide-representation: '{num_tags:dec} entries of type {tags_type}'
+    seq:
+      - id: num_tags
+        type: s4
+      - id: tags
+        type: s4
+        repeat: expr
+        repeat-expr: num_tags
+    instances:
+      tags_type:
+        value: 'tag::tag_int'
+  tag_long_array:
+    -webide-representation: '{num_tags:dec} entries of type {tags_type}'
+    seq:
+      - id: num_tags
+        type: s4
+      - id: tags
+        type: s8
+        repeat: expr
+        repeat-expr: num_tags
+    instances:
+      tags_type:
+        value: 'tag::tag_long'
+enums:
+  tag:
+    0: tag_end
```

Comment:
How about removing the prefix?

Reply:
> How about removing the prefix?

Sure, a good idea. I was afraid that using enum keys like `byte`, `int`, `long`, etc. would cause clashes with statically-typed target language keywords, but I've just checked all of them and surprisingly, it only caused a problem in Lua with keyword `end` (https://github.com/kaitai-io/kaitai_struct/issues/90#issuecomment-766440975). But it's a bug in KSC, not this format, and it can be fixed without having to mangle identifiers (you just replace `end` to `["end"]`). So let's go with it.

Reply:
@KOLANICH Solved in https://github.com/kaitai-io/kaitai_struct_formats/pull/404/commits/3f45153a0c9698f03e3ec92f8e284744ff03ab93.